File: dota2-senate.java
import java.util.ArrayDeque;


class Solution {
    public String predictPartyVictory(String senate) {
        ArrayDeque<Integer> r = new ArrayDeque<>(), d = new ArrayDeque<>();
        int index = 0;
        for (; index < senate.length(); index++) {
            char value = senate.charAt(index);
            if (value == 'R'){
                r.add(index);
            } else {
                d.add(index);
            }
        }

        while (!r.isEmpty() && !d.isEmpty()) {
            if (r.peek() < d.peek()) {
                r.add(index++);
            } else {
                d.add(index++);
            }
            r.removeFirst();
            d.removeFirst();
        }

        if (r.isEmpty()) {
            return "Dire";
        } else {
            return "Radiant";
        }
    }

    // public static void main(String[] args) {
    //     Solution s = new Solution();
    //     System.out.println(s.predictPartyVictory("DRR"));
    // }
}

File: dota2-senate.py
from collections import deque


class Solution:

    def predictPartyVictory(self, senate: str) -> str:
        r = deque()
        d = deque()
        for index, value in enumerate(senate):
            if value == "R":
                r.append(index)
            else:
                d.append(index)
        index += 1

        

        while len(r) and len(d):
            if r[0] < d[0]:
                r.append(index)
            else:
                d.append(index)
            r.popleft()
            d.popleft()
            index += 1
        
        if len(r):
            return "Radiant"
        else:
            return "Dire"
        
print(Solution().predictPartyVictory("RDD"))